pennylane/templates/layers/gate_fabric.py

```diff
@@ -0,0 +1,222 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Contains the quantum number preserving GateFabric template.
+"""
+# pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+import numpy as np
+import pennylane as qml
+from pennylane.operation import Operation, AnyWires
+
+
+class GateFabric(Operation):
+    r"""Implements a local, expressive, and quantum-number-preserving ansatz using VQE circuit fabrics
+    proposed by Anselmetti *et al.* in `arXiv:2104.05692 <https://arxiv.org/abs/2104.05695>`_.
+
+    This template prepares the :math:`N` qubits trial state by applying :math:`D` layers of gate fabric block
+    :math:`\hat{U}_{GF}(\vec{\theta},\vec{\phi})` to the Hartree-Fock state in the Jordan-Wigner basis
+
+    .. math::
+
+        \vert \Psi(\vec{\theta},\vec{\phi})\rangle =
+        \hat{U}_{GF}^{(D)}(\vec{\theta}_{D},\vec{\phi}_{D}) \ldots
+        \hat{U}_{GF}^{(2)}(\vec{\theta}_{2},\vec{\phi}_{2})
+        \hat{U}_{GF}^{(1)}(\vec{\theta}_{1},\vec{\phi}_{1}) \vert HF \rangle
+
+    where each of the gate fabric block :math:`\hat{U}_{GF}(\vec{\theta},\vec{\phi})` comprises of 2-parameter 4-qubit
+    gate elements :math:`\hat{Q}(\theta, \phi)` that act on 4-nearest-neighboring qubits. The circuit implementing a
+    single layer of the gate fabric block for :math:`N = 8` is shown in the figure below:
+
+    .. figure:: ../../_static/templates/layers/gate_fabric_layer.png
```

Comment:
A lot of the images we've used for qchem templates are boring black and white and look worse than the images for other templates. Can we try to add some colour to this to replicate other templates?

![image](https://user-images.githubusercontent.com/40441298/135154869-67f4bea2-fd1d-4c81-9556-3825a6fec6b0.png)